src/coreclr/src/jit/lsra.cpp

```diff
@@ -5523,42 +5656,49 @@ void LinearScan::allocateRegisters()
             continue;
         }
 
-        // If this is a FixedReg, disassociate any inactive constant interval from this register.
-        // Otherwise, do nothing.
-        if (refType == RefTypeFixedReg)
+        if (currentRefPosition->isPhysRegRef)
         {
             RegRecord* regRecord        = currentRefPosition->getReg();
             Interval*  assignedInterval = regRecord->assignedInterval;
 
-            if (assignedInterval != nullptr && !assignedInterval->isActive && assignedInterval->isConstant)
+            updateNextFixedRef(regRecord, currentRefPosition->nextRefPosition);
```

Comment:
curious, shouldn't this be under the `refType == RefTypeFixedReg` check? Or do we consider every physReg as FixedRef?

Reply:
We consider both `RefTypeFixedReg` and `RefTypeKill`s as FixedRef, which is probably confusing so perhaps it should be "NextPhysRef" instead, or something like that.